### ISSUE TITLE
Uncaught TypeError: Argument 1 passed to Magento\Catalog\Block\Product\ImageFactory::getRatio() must be of the type integer

### ISSUE THREAD

**Issue by rr10** [magento/magento2#27913]
0

i have this error in product page after upgrade to magento 2.3.4 from 2.2.6 php 7.2

magento 2.3.4
php 7.2
centos
apache
template porto 2.3.2

```
[20-Apr-2020 15:39:48 UTC] PHP Fatal error: Uncaught TypeError: Argument 1 passed to Magento\Catalog\Block\Product\ImageFactory::getRatio() must be of the type integer, null given, called in /home/rlk001/public_html/vendor/magento/module-catalog/Block/Product/ImageFactory.php on line 172 and defined in /home/rlk001/public_html/vendor/magento/module-catalog/Block/Product/ImageFactory.php:105 Stack trace:
0 /home/rlk001/public_html/vendor/magento/module-catalog/Block/Product/ImageFactory.php(172): Magento\Catalog\Block\Product\ImageFactory->getRatio(NULL, NULL)
1 /home/rlk001/public_html/generated/code/Magento/Catalog/Block/Product/ImageFactory/Interceptor.php(24): Magento\Catalog\Block\Product\ImageFactory->create(Object(Magento\Catalog\Model\Product\Interceptor), 'product_page_im...', Array)
2 /home/rlk001/public_html/vendor/magento/module-catalog/Block/Product/ImageBuilder.php(135): Magento\Catalog\Block\Product\ImageFactory\Interceptor->create(Object(Magento\Catalog\Model\Product\Interceptor), 'product_page_im...', Array)
3 /home/rlk001/public_ht in /home/rlk001/public_html/vendor/magento/module-catalog/Block/Product/ImageFactory.php on line 105enter code here

```
thanks

i have found a similar issue
[https://magento.stackexchange.com/questions/264084/uncaught-typeerror-argument-1-passed-to-magento-catalog-block-product-imagefact](url)

**Comment by mrtuvn** [comment]
@rr10 This issue fixed in this PR https://github.com/magento/magento2/pull/26974

**Comment by engcom-Alfa** [comment]
Hi @rr10 .

Looks like this issue has already been fixed in PR #26974.

So, we have to close it.
Please feel free to comment, reopen or create new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)
if you are still facing this issue on the latest `2.4-develop` branch. Thank you for collaboration.
